### ISSUE TITLE
Test suite is missing test coverage for NaN canonicalizing JS engine + LLVM opts.

### ISSUE THREAD

**Issue by juj** [emscripten-core/emscripten#3742]
Currently our SIMD suite has tests for various SIMD code that expects NaN canonicalization to not to occur in SIMD registers. We have the `default` suite, which tests "SpiderMonkey with no asm.js" + "LLVM with no opts", and then we have the other optimized suites, which test "SpiderMonkey with asm.js" + "LLVM with opts". Since SM doesn't canonicalize NaNs in scalar registers in asm.js, we are missing detecting in our SIMD test suite any LLVM optimizer bugs that cause wrong canonicalization, i.e. "a NaN-canonicalizing JS engine" + "LLVM with opts".

In other words, when we enable LLVM opts, we coincidentally enable running in asm.js, so we don't canonicalize anymore. To fix this, we should run the SIMD tests with LLVM opts enabled, but with `"use asm";` disabled, so that we can detect NaN canonicalization issues that occur from LLVM optimizations. I tried doing a `-s ASM_JS=0 -O2` build, but that is currently not allowed. How should we best add coverage to test this?

**Comment by kripken** [comment]
Building with `-s ASM_JS=2` will make it emit "almost asm" instead of "use asm".

That happens automatically in `-O0`, so maybe we do already have testing?

**Comment by juj** [comment]
`-s ASM_JS=2` sounds good, I'll work on adding that.
